File: db/java/app/src/main/java/com/example/sales/domain/model/Company.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 取引先マスタのEntityクラス
 */
public class Company {
    private String companyCode;
    private String companyName;
    private String companyNameKana;
    private Integer supplierType;
    private String zipCode;
    private String state;
    private String address1;
    private String address2;
    private Integer noSalesFlag;
    private Integer wideUseType;
    private String companyGroupCode;
    private Integer maxCredit;
    private Integer tempCreditUp;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public Company() {}

    // Getter/Setter
    public String getCompanyCode() { return companyCode; }
    public void setCompanyCode(String companyCode) { this.companyCode = companyCode; }

    public String getCompanyName() { return companyName; }
    public void setCompanyName(String companyName) { this.companyName = companyName; }

    public String getCompanyNameKana() { return companyNameKana; }
    public void setCompanyNameKana(String companyNameKana) {
        this.companyNameKana = companyNameKana;
    }

    public Integer getSupplierType() { return supplierType; }
    public void setSupplierType(Integer supplierType) { this.supplierType = supplierType; }

    public String getZipCode() { return zipCode; }
    public void setZipCode(String zipCode) { this.zipCode = zipCode; }

    public String getState() { return state; }
    public void setState(String state) { this.state = state; }

    public String getAddress1() { return address1; }
    public void setAddress1(String address1) { this.address1 = address1; }

    public String getAddress2() { return address2; }
    public void setAddress2(String address2) { this.address2 = address2; }

    public Integer getNoSalesFlag() { return noSalesFlag; }
    public void setNoSalesFlag(Integer noSalesFlag) { this.noSalesFlag = noSalesFlag; }

    public Integer getWideUseType() { return wideUseType; }
    public void setWideUseType(Integer wideUseType) { this.wideUseType = wideUseType; }

    public String getCompanyGroupCode() { return companyGroupCode; }
    public void setCompanyGroupCode(String companyGroupCode) {
        this.companyGroupCode = companyGroupCode;
    }

    public Integer getMaxCredit() { return maxCredit; }
    public void setMaxCredit(Integer maxCredit) { this.maxCredit = maxCredit; }

    public Integer getTempCreditUp() { return tempCreditUp; }
    public void setTempCreditUp(Integer tempCreditUp) { this.tempCreditUp = tempCreditUp; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
}

File: db/go/sales_management/internal/model/company.go
package model

import "time"

// Company は取引先マスタのドメインモデルです
// すべての取引先の基本情報を保持する
// 顧客、仕入先などの役割に応じた詳細情報は別テーブルで管理
type Company struct {
	CompanyCode          string    `db:"取引先コード"`
	CompanyName          string    `db:"取引先名"`
	CompanyNameKana      string    `db:"取引先名カナ"`
	SupplierCategory     int       `db:"仕入先区分"`
	PostalCode           string    `db:"郵便番号"`
	Prefecture           string    `db:"都道府県"`
	Address1             string    `db:"住所１"`
	Address2             string    `db:"住所２"`
	TransactionProhibit  int       `db:"取引禁止フラグ"`
	MiscCategory         int       `db:"雑区分"`
	CompanyGroupCode     string    `db:"取引先グループコード"`
	CreditLimit          int       `db:"与信限度額"`
	TemporaryCreditBoost int       `db:"与信一時増加枠"`
	CreatedAt            time.Time `db:"作成日時"`
	CreatedBy            string    `db:"作成者名"`
	UpdatedAt            time.Time `db:"更新日時"`
	UpdatedBy            string    `db:"更新者名"`
}
